From ba2c5c07394fc2aa29c258a91ec61b859c303337 Mon Sep 17 00:00:00 2001
From: Xiang Xu <xuxiang@eswincomputing.com>
Date: Tue, 12 Aug 2025 16:09:28 +0800
Subject: [PATCH 462/540] feat:implement pin header function of ce1

Changelogs:
Modify dts in order to implement pin header functions of ce1

Signed-off-by: Xiang Xu <xuxiang@eswincomputing.com>
---
 arch/riscv/boot/dts/eswin/eic7700-ce1.dtsi | 119 +++++++++++++++++----
 1 file changed, 97 insertions(+), 22 deletions(-)

diff --git a/arch/riscv/boot/dts/eswin/eic7700-ce1.dtsi b/arch/riscv/boot/dts/eswin/eic7700-ce1.dtsi
index 1d8bff47aa8f..79900c55804f 100644
--- a/arch/riscv/boot/dts/eswin/eic7700-ce1.dtsi
+++ b/arch/riscv/boot/dts/eswin/eic7700-ce1.dtsi
@@ -583,21 +583,11 @@ &d0_uart2 {
 	status = "okay";
 };
 
-&d0_uart3 {
-	/* unused */
-	status = "disabled";
-};
-
 &d0_uart4 {
 	/* unused */
 	status = "disabled";
 };
 
-&ssi0 {
-	/* pin header */
-	status = "okay";
-};
-
 &ssi1 {
 	/* unused */
 	status = "disabled";
@@ -715,11 +705,6 @@ pcf_rtc:rtc@51 {
 	};
 };
 
-&d0_i2c2 {
-	/* pin header */
-	status = "okay";
-};
-
 &d0_i2c3 {
 	/* mipi csi0/csi1 & mipi dsi fpc? */
 	status = "okay";
@@ -755,11 +740,6 @@ &d0_i2c9 {
 	status = "disabled";
 };
 
-&d0_aon_i2c0 {
-	/* pin header */
-	status = "okay";
-};
-
 &d0_aon_i2c1 {
 	/* ina226x3 */
 	status = "okay";
@@ -838,10 +818,55 @@ &timer3 {
 
 &pinctrl {
 	status = "okay";
-};
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio18_default &pinctrl_gpio19_default &pinctrl_gpio20_default &pinctrl_gpio21_default
+			&pinctrl_gpio39_default &pinctrl_gpio40_default &pinctrl_gpio70_default &pinctrl_gpio73_default
+			&pinctrl_gpio74_default &pinctrl_gpio75_default &pinctrl_gpio76_default &pinctrl_gpio79_default
+			&pinctrl_gpio80_default &pinctrl_gpio81_default &pinctrl_gpio82_default &pinctrl_gpio85_default
+			&pinctrl_gpio87_default >;
+
+    /* pin header default function for GPIO
+       SPI1 (CS0,SCLK,MOSI,MISO,D2,D3,CS1): GPIO 35,36,37,38,39,40,41
+       I2C2 (SCL,SDA): GPIO 48,49
+	   I2C10 (SCL,SDA): GPIO 102,103
+       UART3(TX,RX): GPIO 92,93
+	   I2S0 (BCLK, WCLK, SDI, SDO) : GPIO 18, 19, 20, 21
+
+    */
+};
+/******** pin headers *********
+     pin : function
+       3 : I2C1 SCL
+       5 : I2C1 SDA
+       7 : GPIO82
+       8 : UART3 TX
+      10 : UART3 RX
+      11 : GPIO76
+      12 : GPIO18
+      13 : GPIO85
+      15 : GPIO81
+      16 : GPIO79
+      18 : GPIO39
+      19 : SPI1 D0
+      21 : SPI1 D1
+      22 : GPIO40
+      23 : SPI1 CLK
+      24 : SPI1 CS0
+      26 : SPI1 CS1
+      27 : I2C4 SDA
+      28 : I2C4 SCL
+      29 : GPIO80
+      31 : GPIO70
+      32 : GPIO73 (x)
+      33 : GPIO74
+      35 : GPIO19
+      36 : GPIO87 (x)
+      37 : GPIO75
+      38 : GPIO20
+      40 : GPIO21
+******************************/
 
 /* GPIO Function Description
-
 	gpio0   : pcie prstn(I)
 	gpio5   : led back light power on/off(O)
 	gpio6   : system key(I)
@@ -877,3 +902,53 @@ &dev_llc_d0{
 	npu-supply=<&npu_vcc1>;
 	status = "okay";
 };
+
+/* pin header */
+&d0_uart3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart3_default>;
+	status = "okay";
+};
+
+&ssi0 {
+	/* pin header */
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_dual_spi1_default>;
+	num-cs = <1>;
+	spi-flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		spi-max-frequency = <4800000>;
+		rx-sample-delay-ns = <10>;
+	};
+};
+
+&d0_i2c2 {
+	/* pin header */
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c2_default>;
+	tmp102@48 {
+		compatible = "ti,tmp102";
+		reg = <0x48>;
+		label = "board_tmp";
+		#thermal-sensor-cells = <1>;
+	};
+};
+
+&d0_aon_i2c0 {
+	/* pin header */
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c10_default>;
+	sys_power:ina226@40 {
+		compatible = "ti,ina226";
+		#io-channel-cells = <1>;
+		label = "sys_power";
+		reg = <0x40>;
+		shunt-resistor = <1000>;
+	};
+};
-- 
2.54.0

